compiler/qsc_frontend/src/resolve.rs

```diff
@@ -566,10 +712,236 @@ impl Resolver {
+                ) => {
+                    self.errors.push(Error::ExportedExternalItem(item.span()));
+                    continue;
+                }
+                Res::Item(id, _) => id,
+                _ => {
+                    self.errors.push(Error::ExportedNonItem(item.span()));
+                    continue;
+                }
+            };
+
+            let mut maybe_err = None;
+            match term_or_ty {
+                NameKind::Ty => {
+                    if let Some(ScopeItemEntry {
+                        source: ItemSource::Exported,
+                        ..
+                    }) = scope.tys.get(&item.name().name)
+                    {
+                        maybe_err = Some(Error::DuplicateExport(
+                            item.path.name.name.to_string(),
+                            item.path.span,
+                        ));
+                    }
+                    scope.tys.insert(
+                        Rc::clone(&item.name().name),
+                        ScopeItemEntry::new(resolved_item_id, ItemSource::Exported),
+                    );
+                }
+                NameKind::Term => {
+                    if let Some(ScopeItemEntry {
+                        source: ItemSource::Exported,
+                        ..
+                    }) = scope.terms.get(&item.name().name)
+                    {
+                        maybe_err = Some(Error::DuplicateExport(
+                            item.path.name.name.to_string(),
+                            item.path.span,
+                        ));
+                    }
+                    scope.terms.insert(
+                        Rc::clone(&item.name().name),
+                        ScopeItemEntry::new(resolved_item_id, ItemSource::Exported),
+                    );
+                }
+            };
+
+            if let Some(err) = maybe_err {
+                self.errors.push(err);
+            }
+
+            if let Some(namespace) = current_namespace {
+                match term_or_ty {
+                    NameKind::Ty => {
+                        self.globals.tys.get_mut_or_default(namespace).insert(
+                            Rc::clone(&item.name().name),
+                            Res::Item(resolved_item_id, ItemStatus::Available),
+                        );
+                    }
+                    NameKind::Term => {
+                        self.globals.terms.get_mut_or_default(namespace).insert(
+                            Rc::clone(&item.name().name),
+                            Res::Item(resolved_item_id, ItemStatus::Available),
+                        );
+                    }
+                }
+            }
+
+            self.names.insert(item.path.id, resolved_item);
+        }
+    }
+
+    /// This function performs the logic to see if a path is a ty, term, or namespace, and then
+    /// performs the appropriate logic to bind the path.
+    fn bind_exported_path_of_unknown_name_kind(
```

Comment:
Isn't this method doing basically the same thing as `bind_import`? Can we make them into one method?

In general, I can't understand why export and import implementations are so divergent in this PR, when an `export` is basically the same thing as an `import` (with a few minor differences like `ItemSource` and lack of support for globs). The resolution and binding to be almost exactly the same code in both, no? Is there any conceivable case where an `import`ed item should successfully resolve, but a re-`export` of the same shouldn't? Or vice versa?